hackathon_site/event/serializers.py

```diff
@@ -85,6 +85,26 @@ def update(self, instance: Profile, validated_data):
         # ProfileSerializer update function and we want to keep the logic between these two update functions separate.
         return serializers.ModelSerializer.update(self, instance, validated_data)
 
+    def create(self, validated_data):
+        try:
+            self.context["request"].user.profile
```

Comment:
is this suppose to be an if statement?

Reply:
performing a try catch like this is a bit problematic, I'd prefer if you'd just use an if statement: if user.profile == None then raise exception

Reply:
Hey Yuying! I didn't notice the test wasn't passing, sorry! I reran the code after talking to Leo and Mustafa, and apparently if you run the tests after midnight, then an error occurs (for the email test). If the test that you got the 500 error on was different though😓

Reply:
In addition, with regards to the if else statement, I tried replacing it with `self.context["request"].user.profile == None`, but the issue was that if the profile isn't created, the "profile" attribute doesn't exist, and thus the code throws a  "RelatedObjectDoesNotExist" error. Similarly if I try `if user.profile == None`, I get a NameError, because the name 'user' doesn't exist.

Reply:
sorry @terryluan12, my python not so good atm the correct condition is `hasattr(self.context["request"].user, 'profile')` this will not throw any errors.

Reply:
Could you address all the PR comments first and I'll check the 500 error problem again